### ISSUE TITLE
`disallowed_methods` false positive `instant::Instant` vs `std::time::Instant`

### ISSUE THREAD

**Issue by emilk** [rust-lang/rust-clippy#10406]
### Summary

I want to forbid `std::time::Instant::now`, because it doesn't work on web/wasm:

```
❯ cat clippy.toml 
disallowed-methods = ["std::time::Instant::now"]
```

Instead I want to use https://crates.io/crates/instant

However, when I use `instant::Instant::now` I get a false-positive error.

### Lint Name

disallowed_methods

### Reproducer

I tried this code:

```rust
fn main() {
    instant::Instant::now();
}
```

I saw this happen:

```
warning: use of a disallowed method `std::time::Instant::now`
 --> src/main.rs:2:15
  |
2 |     let now = instant::Instant::now();
  |               ^^^^^^^^^^^^^^^^^^^^^^^
  |
```

I expected to see this happen: No warning

### Version

```text
rustc 1.67.1 (d5a82bbd2 2023-02-07)
binary: rustc
commit-hash: d5a82bbd26e1ad8b7401f6a718a9c57c96905483
commit-date: 2023-02-07
host: aarch64-apple-darwin
release: 1.67.1
LLVM version: 15.0.6
```

**Comment by Jarcho** [comment]
This may not be fixable. `instant::Instant` is an alias to `std::time::Instant` on everything but wasm.

**Comment by emilk** [comment]
Oh right, of course… that’s interesting. I guess I would need to only disallow `std::time::Instant` on wasm32, which isn’t possible afaict. Or make a PR to `instant` to turn `instant::Instant` into a thin newtype instead. Thanks for the explanation!
